alibi_detect/utils/pytorch/kernels.py

```diff
@@ -29,12 +36,76 @@ def sigma_median(x: torch.Tensor, y: torch.Tensor, dist: torch.Tensor) -> torch.
     return sigma
 
 
-class GaussianRBF(nn.Module):
+class BaseKernel(nn.Module):
+    """
+    The base class for all kernels.
+    """
+    def __init__(self) -> None:
+        super().__init__()
+        self.parameter_dict: dict = {}
+        self.active_dims: Optional[list] = None
+        self.feature_axis: int = -1
+
+    def forward(self, x: torch.Tensor, y: torch.Tensor, infer_parameter: bool = False) -> torch.Tensor:
+        raise NotImplementedError
+
+
+class SumKernel(nn.Module):
+    """
+    Construct a kernel by averaging two kernels.
+
+    Parameters:
+    ----------
+        kernel_a
+            the first kernel to be summed.
+        kernel_b
+            the second kernel to be summed.
+    """
     def __init__(
         self,
-        sigma: Optional[torch.Tensor] = None,
-        init_sigma_fn: Callable = sigma_median,
-        trainable: bool = False
+        kernel_a: BaseKernel,
+        kernel_b: BaseKernel
+    ) -> None:
+        super().__init__()
+        self.kernel_a = kernel_a
+        self.kernel_b = kernel_b
+
+    def forward(self, x: torch.Tensor, y: torch.Tensor,  infer_parameter: bool = False) -> torch.Tensor:
+        return (self.kernel_a(x, y, infer_parameter) + self.kernel_b(x, y, infer_parameter)) / 2
```

Comment:
Nit pick but the `/2` is unexpected behaviour under the name `SumKernel`. I like having the `/2` as it ensures upper bounds remain known as kernels are added, but perhaps we name as `MeanKernel` or `AverageKernel` instead?

Reply:
Average kernel sounds better.